### PR TITLE
client: add feeHistory method

### PR DESCRIPTION
**Pull request by g11tech**
Extracted from shandong 
- https://github.com/ethereumjs/ethereumjs-monorepo/pull/2316
-
Add the missing methods for faucet/wallets to price and submit transactions

### CODE REVIEW

**Comment by holgerd77** [comment]
As mentioned by @acolytec3 in the chat we should not merge `feeHistory` as is since implementation was only rudimentary, also `gasPrice` is being worked on right now at #2396, will mark this as "blocked" for now to avoid accidental merge.

File: packages/client/lib/rpc/modules/eth.ts [hunk 1]
```diff
@@ -986,4 +989,37 @@ export class Eth {
...
+  async gasPrice() {
+    let gasPrice: bigint = BigInt(0)
+    const latest = await this._chain.getCanonicalHeadHeader()
+    if (this._vm !== undefined && this._vm._common.isActivatedEIP(1559)) {
+      gasPrice = latest.calcNextBaseFee()
+    } else {
+      if (latest.gasUsed > BigInt(0)) gasPrice = latest.gasUsed
+      else {
+        while (gasPrice === BigInt(0)) {
+          const nextBlock = await this._chain.getBlock(latest.number - BigInt(1))
```
Comment (monstermax) [inline packages/client/lib/rpc/modules/eth.ts:1018]: are you looping always on the same block ?

Comment (acolytec3) [reply packages/client/lib/rpc/modules/eth.ts:1018]: This is out of date. We merged a separate PR recently for `eth_gasPrice`.

Comment (holgerd77) [reply packages/client/lib/rpc/modules/eth.ts:1018]: So to add: see comment here https://github.com/ethereumjs/ethereumjs-monorepo/pull/2413#issuecomment-1305497686

**Comment by acolytec3** [comment]
Closing as `gasPrice` has been implemented separately and the implementation of `feeHistory` here is just a stub and should be properly reworked from first principles in a separate PR.

**Pull request closed without merging**